components/net/filemanager_thread.rs

```diff
@@ -779,7 +779,10 @@ pub fn fetch_file_in_chunks(
                     return;
                 }
                 let length = {
-                    let buffer = reader.fill_buf().unwrap().to_vec();
+                    let buffer = match reader.fill_buf() {
+                        Ok(buf) => buf.to_vec(),
+                        Err(_) => vec![],
```

Comment:
Shouldn't we have a way of bubbling this error state out?

Reply:
With this change, an `Unsupported content type: application/octet-stream` error or something similar is displayed as text in the window.

Reply:
How?

Reply:
This is what I get if I try opening a directory:

![servo](https://user-images.githubusercontent.com/30772943/59285797-39c94780-8c77-11e9-90a4-78d37b5867e5.png)